core/src/rpc.rs

```diff
@@ -421,21 +402,33 @@ impl JsonRpcRequestProcessor {
         &self,
         signatures: Vec<Signature>,
         commitment: Option<CommitmentConfig>,
-    ) -> Result<Vec<Option<RpcTransactionStatus>>> {
+    ) -> RpcResponse<Vec<Option<RpcTransactionStatus>>> {
         let mut statuses: Vec<Option<RpcTransactionStatus>> = vec![];
 
         let bank = self.bank(commitment);
 
         for signature in signatures {
             let status = bank.get_signature_confirmation_status(&signature).map(
-                |SignatureConfirmationStatus { slot, status, .. }| RpcTransactionStatus {
+                |SignatureConfirmationStatus {
+                     slot,
+                     status,
+                     confirmations,
+                 }| RpcTransactionStatus {
                     slot,
                     status,
+                    confirmations: if confirmations < MAX_LOCKOUT_HISTORY {
+                        Some(confirmations)
+                    } else {
+                        None
```

Comment:
It'd be nice to have test coverage for this case

Reply:
That would be nice, but the test utility here is a bear to work with. BankForks is built by hard 😬 Since we think this is likely to change in the *very* near future, I'd like to let this slide for now.